addons/escoria-core/game/esc_inputs_manager.gd

```diff
@@ -159,13 +179,70 @@ func clear_stack():
 	hover_stack = []
 
 
+# Unsets the hovered node.
+#
+# **Parameters**
+#
+# - item: the item that was unfocused (mouse_exited)
+func unset_hovered_node(item: ESCItem):
+	if _hovered_element == item:
+		_hovered_element.mouse_exited()
+		_hovered_element = null
+		hotspot_focused = ""
+		
+
+# Sets the hovered node and calls its mouse_entered() method if it was the top
+# most item in hover_stack.
+#
+# #### Parameters
+#
+# - item: the item that was focused (mouse_entered)
+#
+# **Returns**
+# True if item is the new top hovered object
+func set_hovered_node(item: ESCItem) -> bool:
+	if _hovered_element == item \
+			or not escoria.action_manager.is_object_actionable(item.global_id) \
+			or (item is ESCPlayer and not (item as ESCPlayer).selectable):
+		return true
+	
+	if _hovered_element == null:
+		_hovered_element = item
+		_hovered_element.mouse_entered()
+		return true
+ 
+	var hovered_position = 0
+	var i = 0
+	for node in hover_stack:
+		if node == _hovered_element:
+			hovered_position = i
+			break
+		i += 1
+
+	var itemPosition = 0
+	i = 0
+	for node in hover_stack:
+		if node == item:
+			itemPosition = i
+			break
+		i += 1
+
+	if itemPosition >= hovered_position:
+		_hovered_element = item
+		_hovered_element.mouse_entered()
+		return true
+	else:
+		return false
+
+
 # The background was clicked with the LMB
 #
 # #### Parameters
 #
 # - position: Position of the click
 func _on_left_click_on_bg(position: Vector2) -> void:
-	if input_mode == INPUT_ALL and hotspot_focused.empty():
+	if input_mode == INPUT_ALL: # and hotspot_focused.empty():
+		hotspot_focused = ""
```

Comment:
Possible to remove this `hotspot_focused` attribute since we now have `_hovered_element`?